src/custom/utils/index.ts

```diff
@@ -116,3 +116,43 @@ export function formatOrderId(orderId: string): string {
   // 0x is at index 0 of orderId, shorten. Else return id as is
   return has0x?.index === 0 ? shortenOrderId(orderId, 2, orderId.length) : orderId
 }
+
+export async function fetchData(endpoint: string, options?: RequestInit) {
+  let response
+  try {
+    response = await fetch(endpoint, options)
+    if (!response.ok) throw new Error('[fetchData]::Server returned non-ok response.')
+  } catch (error) {
+    console.error(error)
+    throw new Error(error)
+  }
+  return response
+}
+
+export async function promiseFirstResolved<T>(
```

Comment:
It took me a bit to understand the logic. It would help a description message

A comment would help. I see the argument is some kind of promise generators, not promiseList. See how you invoke `promise()`